virtual_machine.go

```diff
@@ -643,3 +643,20 @@ func (v *VirtualMachine) ConvertToTemplate(ctx context.Context) (task *Task, err
 	}
 	return NewTask(upid, v.client), nil
 }
+
+func (v *VirtualMachine) UnmountCloudInitISO(ctx context.Context, device string) error {
+	if v.HasTag(MakeTag(TagCloudInit)) {
+		if ok, err := v.deleteCloudInitISO(ctx); err != nil || !ok {
+			return err
+		}
+		task, err := v.Config(ctx, VirtualMachineOption{
```

Comment:
you should unmount the drive before deleting the ISO, flip these two calls around

Reply:
right